### PR TITLE
Delete dead code in qwen-vl series

### PR DESCRIPTION
**Pull request by zucchini-nlp**
# What does this PR do?

as per title:

- `rotary_pos_emb` is not accessed in forward and is actually a duplicate of `position_embeddings`
- `rope_deltas` also not accessed in forward and duplicates an instance attr `self.rope_deltas`
- returned ModelOutput doesn't need to have `output.rope_deltas` because it is fixed and also can be obtained from `model.base_mode.rope_deltas`
-

### CODE REVIEW

**Comment by zucchini-nlp** [comment]
cc @vasqu , this is ready now whenever you have time. Some light clean-up of unused code in qwen models

File: docs/source/en/model_doc/pe_audio.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ specific language governing permissions and limitations under the License.
 rendered properly in your Markdown viewer.
 
 -->
-*This model was released on {release_date} and added to Hugging Face Transformers on 2025-12-16.*
+*This model was released on 2025-04-17 and added to Hugging Face Transformers on 2025-12-16.*
```
Comment (vasqu) [inline docs/source/en/model_doc/pe_audio.md:16]: Can you look up when this was actually added or revert? Pretty sure it was just recently, max dec 2025

File: src/transformers/models/glm4v_moe/modeling_glm4v_moe.py [hunk 2]
```diff
@@ -659,7 +648,6 @@ def forward(
         self,
         hidden_states: torch.Tensor,
         cu_seqlens: torch.Tensor,
-        rotary_pos_emb: torch.Tensor | None = None,
```
Comment (vasqu) [inline src/transformers/models/glm4v_moe/modeling_glm4v_moe.py:662]: I think we are missing a few `deprecate_kwarg` here

File: src/transformers/models/qwen2_5_vl/modular_qwen2_5_vl.py
Comment (vasqu) [inline src/transformers/models/qwen2_5_vl/modular_qwen2_5_vl.py:1]: Seems like some inheritance is not there so some decorators are missing now, e.g. the deprecator one

File: src/transformers/models/qwen2_5_vl/modeling_qwen2_5_vl.py [hunk 3]
```diff
@@ -226,7 +227,6 @@ def forward(
         self,
         hidden_states: torch.Tensor,
         cu_seqlens: torch.Tensor,
-        rotary_pos_emb: torch.Tensor | None = None,
```
Comment (vasqu) [inline src/transformers/models/qwen2_5_vl/modeling_qwen2_5_vl.py:229]: here it seems

File: src/transformers/models/qwen2_vl/modeling_qwen2_vl.py [hunk 4]
```diff
@@ -393,7 +369,6 @@ def forward(
         self,
         hidden_states: torch.Tensor,
         cu_seqlens: torch.Tensor,
-        rotary_pos_emb: torch.Tensor | None = None,
```
Comment (vasqu) [inline src/transformers/models/qwen2_vl/modeling_qwen2_vl.py:396]: or this might be an issue here instead that it was missing

File: src/transformers/models/qwen2_vl/modeling_qwen2_vl.py [hunk 5]
```diff
@@ -472,14 +447,12 @@ def forward(
...
```
Comment (vasqu) [inline src/transformers/models/qwen2_vl/modeling_qwen2_vl.py:475]: and here

File: src/transformers/utils/auto_docstring.py [hunk 6]
```diff
@@ -4203,6 +4203,13 @@ def auto_class_docstring(cls, custom_intro=None, custom_args=None, checkpoint=No
         doc_class = cls.__doc__
         if custom_args is None and doc_class:
             custom_args = doc_class
+
+        # Pass over docs from the direct parent, if it is a class from `modeling_outputs.py`
```
Comment (vasqu) [inline src/transformers/utils/auto_docstring.py:4207]: Imo, would like this to be a different PR, this seems like it could have a wider impact (and could reduce a few modular files)

**Review by vasqu — COMMENTED** [review]
Some quick comments, I think some deprecate kwarg decorators are still missing (the inheritance is a bit messy tho so likely somewhere from the qwen2-3 vl series)

Other than that, I think the inheritance on auto doc is nice but would isolate that into a different PR

File: docs/source/en/model_doc/pe_audio.md (hunk 1 above)
Comment (zucchini-nlp) [reply docs/source/en/model_doc/pe_audio.md:16]: Yeah, in december, it says " on 2025-12-16.*" :)

This change comes from release data of actual paper on the hub, the date when the paper was published on the hub and is automatically added by "fix-repo"

The wording is weird though and imo it should be smth like "The model paper was published on HF papers on xxxx-xx-xx". I have it long in my TODO to check what happens and fix automatic dates, not the first time seeing random changes on it

File: src/transformers/models/glm4v_moe/modeling_glm4v_moe.py (hunk 2 above)
Comment (zucchini-nlp) [reply src/transformers/models/glm4v_moe/modeling_glm4v_moe.py:662]: I added `deprecate_kwarg` on public classes only on purpose, not sure if we need to do with modules that are not high lvl imported given that `rotary_pos_emb` was never used anyway and will be consumed via kwargs if passed

File: src/transformers/models/qwen2_5_vl/modular_qwen2_5_vl.py
Comment (zucchini-nlp) [reply src/transformers/models/qwen2_5_vl/modular_qwen2_5_vl.py:1]: you mean on `rotary_embeddings`? I answered above

File: src/transformers/utils/auto_docstring.py (hunk 6 above)
Comment (zucchini-nlp) [reply src/transformers/utils/auto_docstring.py:4207]: it is directly linked to the changes here to pass CI, because I am inheriting the `ModelOutput` class and deleted the docstring

File: src/transformers/models/glm4v_moe/modeling_glm4v_moe.py (hunk 2 above)
Comment (vasqu) [reply src/transformers/models/glm4v_moe/modeling_glm4v_moe.py:662]: Imo, even if it's more private modules youd be surprised how many people still use it 😢 it doesnt hurt to add it there for the cushion/security benefit we get

File: docs/source/en/model_doc/pe_audio.md (hunk 1 above)
Comment (vasqu) [reply docs/source/en/model_doc/pe_audio.md:16]: Oh lol I completely overlooked the addition time on transformers but yea good point. Something messy is definitely happening from time to time here

File: src/transformers/utils/auto_docstring.py (hunk 6 above)
Comment (vasqu) [reply src/transformers/utils/auto_docstring.py:4207]: I think current models completely override the docstring in this case to pass ci? 

I'd really just love something that cleans this up after this is merged. Also fine with me in a subsequent PR or at least an issue for us to keep track with this

Comment (zucchini-nlp) [reply src/transformers/utils/auto_docstring.py:4207]: Well the model-specific output classes in `main` are all BaseOutput and all of the docs for those args are already in `auto_docstring`. This PR also starts inheriting `CausalLMOutput` which has fields for loss/logits that are not auto-doc'ed

I see that there are a few classes that overwrite explicitly a doc for `last_hiddens_state`, when it could be deleted and just copied from parent. I can clean it up in subsequent PR after this, wdyt?

Comment (vasqu) [reply src/transformers/utils/auto_docstring.py:4207]: SGTM, that would be awesome

File: src/transformers/models/glm4v_moe/modeling_glm4v_moe.py (hunk 2 above)
Comment (zucchini-nlp) [reply src/transformers/models/glm4v_moe/modeling_glm4v_moe.py:662]: I see your point, added a deprecation message for explicitness and it is still consumed silently via `kwargs`, i.e. no errors downstream

**Comment by zucchini-nlp** [comment]
The tests are passing, so ping for re-review @vasqu

File: src/transformers/models/glm4v/modeling_glm4v.py [hunk 7]
```diff
@@ -366,20 +367,16 @@ def forward(
         self,
         hidden_states: torch.Tensor,
         cu_seqlens: torch.Tensor,
-        rotary_pos_emb: torch.Tensor | None = None,
```
Comment (vasqu) [inline src/transformers/models/glm4v/modeling_glm4v.py:369]: looks like we are missing the decorator here, likely coming from some qwen vl model

Comment (vasqu) [reply src/transformers/models/glm4v/modeling_glm4v.py:369]: Might be coming from `Qwen2_5_VLVisionBlock`

File: src/transformers/models/qwen2_5_omni/modeling_qwen2_5_omni.py [hunk 8]
```diff
@@ -509,32 +514,14 @@ def get_rope_index(
...
-    loss (`torch.FloatTensor` of shape `(1,)`, *optional*, returned when `labels` is provided):
-        Language modeling loss (for next-token prediction).
-    logits (`torch.FloatTensor` of shape `(batch_size, sequence_length, config.vocab_size)`, *optional*):
-        Prediction scores of the language modeling head (scores for each vocabulary token before SoftMax).
-    past_key_values (`Cache`, *optional*, returned when `use_cache=True` is passed or when `config.use_cache=True`):
-        It is a [`~cache_utils.Cache`] instance. For more details, see our [kv cache guide](https://huggingface.co/docs/transformers/en/kv_cache).
-
-        Contains pre-computed hidden-states (key and values in the self-attention blocks) that can be used (see
-        `past_key_values` input) to speed up sequential decoding.
     rope_deltas (`torch.LongTensor` of shape `(batch_size, )`, *optional*):
```
Comment (vasqu) [inline src/transformers/models/qwen2_5_omni/modeling_qwen2_5_omni.py:521]: missing message about deprecation here?

File: src/transformers/models/qwen2_5_vl/modeling_qwen2_5_vl.py [hunk 9]
```diff
@@ -304,20 +305,16 @@ def forward(
...
```
Comment (vasqu) [inline src/transformers/models/qwen2_5_vl/modeling_qwen2_5_vl.py:307]: yea this is the one that is missing I think, all other seem to be modular then

**Review by vasqu — APPROVED** [review]
Just found a few missing ones but other than that should be good then. Let's rerun slow ci just in case?

**Comment by zucchini-nlp** [comment]
run-slow: ernie4_5_vl_moe, exaone4_5, glm46v, glm4v, glm4v_moe, glm_image, glm_ocr, minicpmv4_6, paddleocr_vl, qwen2_5_omni, qwen2_5_vl, qwen2_vl, qwen3_5, qwen3_5_moe, qwen3_omni_moe, qwen3_vl
